expConfig.md

<table border="10">
<tr align="left"><th>No.</th>
<td>C1</td><td>C2</td><td>C3</td><td>C4</td><td>C5</td>
<td>C6</td><td>C7</td><td>C8</td><td>C9</td><td>C10</td>
<td>C11</td><td>C12</td><td>C13</td>
</tr>

<tr align="left"><th>RNN_ACTIVATION</th>
<td>tanh</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td>softmax</td><td>sigmoid</td><td>tanh</td>
</tr>

<tr align="left"><th>volume_normalize</th>
<td>No</td><td>wav_norm</td><td></td><td>No</td><td>wav_norm</td>
<td>No</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
<td>wav_norm</td><td>wav_AMP</td><td>spec_AMP</td>
</tr>

<tr align="left"><th>log_spec_trunc</th>
<td>[-3,5]</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td>[-3,7]</td>
<td>[-3,5]</td><td></td><td></td>
</tr>

<tr align="left"><th>input_size</th>
<td>257</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>output_size</th>
<td>257</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>rnn_size</th>
<td>496</td><td></td><td></td><td></td><td>1024</td>
<td>496</td><td></td><td></td><td>1024</td><td>496</td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>rnn_layers_num</th>
<td>2</td><td></td><td>3</td><td>2</td><td>3</td>
<td>2</td><td></td><td>1</td><td>2</td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>batch_size</th>
<td>128</td><td></td><td></td><td></td><td>64</td>
<td>256</td><td></td><td></td><td>128</td><td>256</td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>learning_rate</th>
<td>0.001</td><td></td><td></td><td></td><td>0.002</td>
<td>0.001</td><td></td><td></td><td></td><td></td>
<td>0.1</td><td>0.001</td><td></td>
</tr>

<tr align="left"><th>min_epoches</th>
<td>10</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>max_epoches</th>
<td>50</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>halving_factor</th>
<td>0.5</td><td></td><td></td><td></td><td></td>
<td>0.7</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>start_halving_impr</th>
<td>0.003</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>end_halving_impr</th>
<td>0.001</td><td></td><td></td><td></td><td></td>
<td>0.0005</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>keep_prob</th>
<td>0.8</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td>0.5</td>
</tr>

<tr align="left"><th>max_grad_norm</th>
<td>5.0</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>model_type</th>
<td>LSTM</td><td></td><td></td><td>BLSTM</td><td>LSTM</td>
<td>BLSTM</td><td>LSTM</td><td>BLSTM</td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>final_train_loss</th>
<td>3.1817</td><td>4.1797</td><td>7.7716</td><td>=</td><td>7.5969</td>
<td>3.0427(0.9128)</td><td>3.8880(1.1664)</td><td>3.2367(0.9710)</td><td>=</td><td>=</td>
<td>(收敛太慢,梯度消失)</td><td>3.1800(0.9540)</td><td></td>
</tr>

<tr align="left"><th>final_validation_loss</th>
<td>5.0099</td><td>6.2518</td><td>7.9592</td><td>=</td><td>7.7481</td>
<td>3.0610(0.9183)</td><td>3.9230(1.1769)</td><td>3.2413(0.9724)</td><td>=</td><td>=</td>
<td>=</td><td>3.2063(0.9619)</td><td></td>
</tr>

<tr align="left">
<th>epoch_duration<br>(Normal)</th>
<td>5.5h</td><td>5.5h</td><td>7.0h</td><td>None</td><td>9.0h</td>
<td>None</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>epoch_duration<br>(SmallTFRecord+tfDataset)</th>
<td>3.0h</td><td>None</td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>epoch_duration<br>(BigTFRecord+tfDataset)</th>
<td>1.5h</td><td>None</td><td></td><td>=</td><td>None</td>
<td>1.8h</td><td>1.2h</td><td>1.3h</td><td>5.9h</td><td>=</td>
<td>=</td><td>1.94h</td><td></td>
</tr>

<tr align="left"><th>epoch_duration<br>(BigTFRecord+tfQueue)</th>
<td>1.9h</td><td>None</td><td></td><td></td><td></td>
<td></td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>speaker num</th>
<td>4</td><td></td><td></td><td></td><td></td>
<td>90</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>train_set speech duration</th>
<td>1126h</td><td></td><td></td><td></td><td></td>
<td>1166h</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th>utterance duration</th>
<td>10S</td><td></td><td></td><td></td><td></td>
<td>3S</td><td></td><td></td><td></td><td></td>
<td></td><td></td><td></td>
</tr>

<tr align="left"><th colspan='120'>表中留空位置表示该参数与其前面的实验参数相同，“=”表示实验尚未得出结果或未进行该实验。<br>
wav_norm: Normlize wav's volume before mixing. Both input and label are normlized.<br>
wav_AMP: Normlize label wav's volume, mixed wav is not normlized. Just label are normlized.<br>
spec_AMP: Normlize label spec's mag, mixed wav's spec is not normlized. Just label are normlized.<br>
</th>
</tr>
</table>


FLAGS.py
import tensorflow as tf


class NNET_PARAM:
  '''
  @decode
  Flag indicating decoding or training.
  Show (mixed_wav, mixed_wav_spec, cleaned_wav, cleaned_wav_pic, cleaned_wav_spec).
  wav_pic is oscillograph.
  wav_spec is spectrum
  '''
  decode = 1
  '''
  @decode_show_more
  Flag indicating show  (label_wav, label_wav_spec, label_wav_pic) or not.
  wav_pic is oscillograph.
  wav_spec is spectrum
  '''
  decode_show_more = 1
  decode_input_norm_speaker_volume = False
  decode_output_norm_speaker_volume = True
  decode_speaker_tracing = True
  # Flag indicating whether to resume training from cptk.
  resume_training = 'false'
  start_epoch = 0
  input_size = 257  # The dimension of input.
  output_size = 257  # The dimension of output per speaker.
  rnn_size = 496  # Number of rnn units to use.
  rnn_num_layers = 2  # Number of layer of rnn model.
  batch_size = 256
  learning_rate = 0.001  # Initial learning rate.
  min_epochs = 10  # Min number of epochs to run trainer without halving.
  max_epochs = 50  # Max number of epochs to run trainer totally.
  halving_factor = 0.7  # Factor for halving.
  # Halving when ralative loss is lower than start_halving_impr.
  start_halving_impr = 0.003
  # Stop when relative loss is lower than end_halving_impr.
  end_halving_impr = 0.0005
  # The num of threads to read tfrecords files.
  num_threads_processing_data = 64
  save_dir = 'exp/lstm_pit'  # Directory to put the train result.
  keep_prob = 0.8  # Keep probability for training dropout.
  max_grad_norm = 5.0  # The max gradient normalization.
  model_type = 'BLSTM'  # BLSTM or LSTM
  LSTM_ACTIVATION = tf.nn.tanh
  # LSTM_ACTIVATION = tf.nn.softmax
  MASK_TYPE = 'IAM'  # IAM: ideal_amplitude_mask; PSM: phase_sensitive_mask.
  # MASK_TYPE='PSM'

  GPU_RAM_ALLOW_GROWTH = True
  USE_MULTIGPU = False  # dont't use multiGPU,because it is not work now...
  GPU_LIST = [0, 2]
  if USE_MULTIGPU:
    if batch_size % len(GPU_LIST) == 0:
      batch_size //= len(GPU_LIST)
    else:
      print('Batch_size %d cannot divided by gpu num %d.' %
            (batch_size, len(GPU_LIST)))
      exit(-1)

  minibatch_size = 400  # batch num to show
  # generate timeline file. # !!! timeling is not work now, so set it false unless to test it.
  time_line = False
  timeline_type = 'minibatch'  # timeline write method. 'epoch' ro 'minibatch'
  RESTORE_PHASE = 'GRIFFIN_LIM'  # 'MIXED','CLEANED','GRIFFIN_LIM'
  GRIFFIN_ITERNUM = 100


class MIXED_AISHELL_PARAM:
  # rawdata, dirs by speakerid, like "....data_aishell/wav/train".
  RAW_DATA = '/home/student/work/pit_test/data'
  DATA_DICT_DIR = '_data/mixed_aishell'
  GENERATE_TFRECORD = False
  PROCESS_NUM_GENERATE_TFERCORD = 64
  SHUFFLE = False

  '''
  TFRECORDS_DIR='/big-data/tmplhf/pit-data/feature_tfrecords_utt03s'
  TFRECORDS_FILE_TYPE='small' # 'big' or 'small'.if 'small', one file per record.
  LEN_WAWE_PAD_TO = 16000*3 # Mixed wave length (16000*3 is 3 seconds)
  UTT_SEG_FOR_MIX = [260, 290]  # Separate utt to [0:260],[260,290],[290:end]
  DATASET_NAMES = ['train', 'validation', 'test_cc']
  DATASET_SIZES = [1400000, 18000, 180000]
  '''
  # TFRECORDS_DIR = '/workspace/alldata/pit-data/feature_tfrecords_utt03s_big' # for docker
  TFRECORDS_DIR = '/ntfs/tmplhf/pit-data/feature_tfrecords_utt03s_big'
  # 'big' or 'small'.if 'small', one file per record.
  TFRECORDS_FILE_TYPE = 'big'
  LEN_WAWE_PAD_TO = 16000*3  # Mixed wave length (16000*3 is 3 seconds)
  UTT_SEG_FOR_MIX = [260, 290]  # Separate utt to [0:260],[260,290],[290:end]
  DATASET_NAMES = ['train', 'validation', 'test_cc']
  DATASET_SIZES = [1400000, 18000, 180000]

  '''
  TFRECORDS_DIR = '/big-data/tmplhf/pit-data/feature_tfrecords_utt10s'
  TFRECORDS_FILE_TYPE='small' # 'big' or 'small'.if 'small', one file per record.
  LEN_WAWE_PAD_TO = 16000*10 # Mixed wave length (16000*3 is 3 seconds)
  UTT_SEG_FOR_MIX = [260, 290]  # Separate utt to [0:260],[260,290],[290:end]
  DATASET_NAMES = ['train', 'validation', 'test_cc']
  DATASET_SIZES = [405600, 5400, 20000]
  '''
  '''
  TFRECORDS_DIR='/ntfs/tmplhf/pit-data/feature_tfrecords_utt10s_big'
  TFRECORDS_FILE_TYPE='big' # 'big' or 'small'.if 'small', one file per record.
  LEN_WAWE_PAD_TO = 16000*10 # Mixed wave length (16000*3 is 3 seconds)
  UTT_SEG_FOR_MIX = [260, 290]  # Separate utt to [0:260],[260,290],[290:end]
  DATASET_NAMES = ['train', 'validation', 'test_cc']
  DATASET_SIZES = [405600, 5400, 20000]
  '''

  # WAVE_NORM=True
  WAVE_NORM = False
  LOG_NORM_MAX = 5
  LOG_NORM_MIN = -3
  NFFT = 512
  OVERLAP = 256
  FS = 16000
